docs/adr/0002-database-corruption.md

```diff
@@ -0,0 +1,132 @@
+* C) What do we do when we identify corrupted databases?
+  * Delete the database file and recreate the database
+  * Move the database file and recreate the database
+  * Have the component fail
+
+## Decision Outcome
+
+* A: Whenever we perform a database operation
+
+This was chosen because it can be a long time between migrations and we want
+to de able to fix corrupt database during that period.
+
+* B: Compare error values against known corruption errors when opening the database
+
+This was chosen because it had the simplest initial implementation and was
+least likely to misidentify errors.  We can target only errors that clearly
+indicate corruption, for example the `DatabaseCorrupt` error code.  We can also
+only target opening the database, which is the easiest time to fix issues.
+
+* C: Delete the database file and recreate the database
+
+This follows the choice of B.  Since we're being conservative in identifying
+errors, we can delete the database file with relative confidence.
+
+### Implementation notes:
+
+This can be implemented by updating the `Store` code to wrap each database
+operation with a check for errors that we can positively identify as
+corruption-related.  If we see one of those errors, then we can delete the
+database file and start from scratch.
+
+Also, we will start tracking Sentry errors and telemetry to identify errors
+that we know are from SQLite corruption.
+
+# Pros and Cons of the Options
+
+### A: Check for corruption at migration time only
+* Good, because corruption errors are more likely to be noticed at migration time.
+* Good, because it's a natural time to delete the database -- since the user hasn't run any queries yet.
+* Bad, migrations happen infrequently.  Errors that manifest during migrations
+  were probably also causing issues before.
+
+### A: Check for corruption whenever we perform a database operation (Decided)
+* Good, because this gives us the opportunity to fix corruption errors sooner.
+* Bad, because the consumer code probably doesn't expect the database to be
+  deleted and recreated in the middle of a query.  However, this is just an
+  extreme case of normal database behavior -- for example any given row can be
+  deleted during a sync.
+
+### B: Assume all errors during migration are from corrupt databases
+* Bad, because creates the possibility of deleting valid user data.
+* Bad, because logic errors are most likely during migrations.
+
+### B: Compare error values against known corruption errors for all operations
+* Good, because should eliminate the possibility of deleting valid user data.
+* Bad, because we don't currently have a good list of these.
+* Bad, because it's tricky to try to repair a database that the consumer code
+  is currently using.
+
+### B: Compare error values against known corruption errors when opening the database (Decided)
+* Good, because should eliminate the possibility of deleting valid user data.
```

Comment:
Not sure I understand what you mean here - is the premise that checking during db open is likely to avoid false-positives, where we think the issue was db corruption but actually it was some logic error causing a bad query?